RIOT-OS/RIOT
# drivers/at86rf215: implement MR-OFDM

### Contribution description

This adds MR-OFDM modulation to the `at86rf215` driver.

This also adds new commands to `ifconfig` to configure the new modulation parameters.

### Testing procedure

Switch to the new PHY mode with `ifconfig 7 set phy_mode mr-ofdm`. Note that the `L2-PDU` has changed to 2022 bytes (802.15.4g can send 2047 byte frames).

```
2020-05-04 03:41:16,173 # Iface  7  HWaddr: 01:57  Channel: 0  NID: 0x23  PHY: MR-OFDM 
2020-05-04 03:41:16,180 #            Option: 1  MCS: 0 (BPSK, rate 1/2, 4x frequency repetition) 
2020-05-04 03:41:16,184 #           Long HWaddr: 1A:F9:8F:5F:30:5D:05:8E 
2020-05-04 03:41:16,191 #            TX-Power: 0dBm  State: IDLE  max. Retrans.: 3  CSMA Retries: 4 
2020-05-04 03:41:16,197 #           AUTOACK  ACK_REQ  CSMA  L2-PDU:2022 MTU:1280  HL:64  RTR  
2020-05-04 03:41:16,199 #           6LO  IPHC  
2020-05-04 03:41:16,202 #           Source address length: 8
2020-05-04 03:41:16,204 #           Link type: wireless
2020-05-04 03:41:16,210 #           inet6 addr: fe80::18f9:8f5f:305d:58e  scope: link  VAL
2020-05-04 03:41:16,213 #           inet6 group: ff02::2
2020-05-04 03:41:16,216 #           inet6 group: ff02::1
2020-05-04 03:41:16,219 #           inet6 group: ff02::1:ff5d:58e
2020-05-04 03:41:16,220 #           
2020-05-04 03:41:16,223 #           Statistics for Layer 2
2020-05-04 03:41:16,226 #             RX packets 0  bytes 0
2020-05-04 03:41:16,230 #             TX packets 4 (Multicast: 4)  bytes 172
2020-05-04 03:41:16,233 #             TX succeeded 4 errors 0
2020-05-04 03:41:16,236 #           Statistics for IPv6
2020-05-04 03:41:16,239 #             RX packets 0  bytes 0
2020-05-04 03:41:16,244 #             TX packets 4 (Multicast: 4)  bytes 256
2020-05-04 03:41:16,247 #             TX succeeded 4 errors 0
2020-05-04 03:41:16,247 # 
2020-05-04 03:41:30,034 #  ping6 fe80::2068:3123:59f5:d23a%7
2020-05-04 03:41:30,053 # 12 bytes from fe80::2068:3123:59f5:d23a%7: icmp_seq=0 ttl=64 rssi=-41 dBm time=12.051 ms
2020-05-04 03:41:31,054 # 12 bytes from fe80::2068:3123:59f5:d23a%7: icmp_seq=1 ttl=64 rssi=-40 dBm time=12.051 ms
2020-05-04 03:41:32,054 # 12 bytes from fe80::2068:3123:59f5:d23a%7: icmp_seq=2 ttl=64 rssi=-41 dBm time=12.051 ms
```

You should be able to configure the OFDM option (`ifconfig 7 set option [1…4]`) as well as the Modulation and Coding Scheme (`ifconfig 7 set mcs [0…6]`):

```
2020-05-04 03:49:00,938 #  ifconfig 7 set mcs 2
2020-05-04 03:49:04,010 #  ifconfig 7 set option 4
2020-05-04 03:52:49,320 #  ping6 fe80::2068:3123:59f5:d23a%7
2020-05-04 03:52:49,348 # 12 bytes from fe80::2068:3123:59f5:d23a%7: icmp_seq=0 ttl=64 rssi=-42 dBm time=19.828 ms
2020-05-04 03:52:50,348 # 12 bytes from fe80::2068:3123:59f5:d23a%7: icmp_seq=1 ttl=64 rssi=-42 dBm time=19.828 ms
2020-05-04 03:52:51,348 # 12 bytes from fe80::2068:3123:59f5:d23a%7: icmp_seq=2 ttl=64 rssi=-42 dBm time=19.827 ms
2020-05-04 03:52:51,348 # 
2020-05-04 03:52:51,353 # --- fe80::2068:3123:59f5:d23a PING statistics ---
2020-05-04 03:52:51,358 # 3 packets transmitted, 3 packets received, 0% packet loss
2020-05-04 03:52:51,362 # round-trip min/avg/max = 19.827/19.827/19.828 ms
```

Nodes can talk to each other if they use the same option, but different Modulation & Coding Schemes.

### Issues/PRs references

split off #12128
requires #13979 to send full 2k frames
